### DISCUSSION TITLE
How to call restic's check

### DISCUSSION THREAD

**Discussion by fdw** [cupcakearmy/autorestic#174] [Q&A] [answered]
Restic has a [`check` command to check a repository's consistency and integrity](https://restic.readthedocs.io/en/stable/045_working_with_repos.html#checking-integrity-and-consistency). [Autorestic's `check`](https://autorestic.vercel.app/cli/check) seems to be something different ("This is mostly an internal command, but useful to verify if a backend is configured correctly.").

Is there a way to call restic's `check` through autorestic?

**Comment by ironicbadger** [comment, marked as answer]
Like this...

```
root@morpheus:~# autorestic exec check -v -b morpheus-documents
Using config file: /root/.autorestic.yml


    Executing on "morpheus-documents"

> Executing: /usr/bin/restic check
using temporary cache in /tmp/restic-check-cache-483506628
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
[0:16] 100.00%  92 / 92 snapshots

no errors were found
```

**Comment by ssanj** [comment]
Is it also possible to pass through the `--read-data` option to `check` as mentione in the Restic [checking-integrity-and-consistency](https://restic.readthedocs.io/en/stable/045_working_with_repos.html#checking-integrity-and-consistency) docs?

I tried several variations but couldn't get it to work.

eg.

```
autorestic -c ./autorestic.yml exec check -v -a --read-data
autorestic -c ./autorestic.yml exec 'check --read-data'  -v -a
```

Is it possible to pass additional arguments to the Restic functionality called through `exec` ?

**Reply by ssanj** [reply]
Happy to move this to a separate question if that helps.
